### PR TITLE
Add -section option to 'req' command

### PR DESCRIPTION
**Pull request by richsalz**
This removes "req" as the hardwired section for the req command.
Doing this will let us merge some test configs which will make fips/non-fips testing easier
It is analogous to the CA "-name" option; I added -section as an option to ca.c

- [X] documentation is added or updated
- [X] tests are added or updated

### CODE REVIEW

File: apps/req.c [hunk 1]
```diff
@@ -109,6 +111,7 @@ const OPTIONS req_options[] = {
     OPT_SECTION("Certificate"),
     {"new", OPT_NEW, '-', "New request"},
     {"config", OPT_CONFIG, '<', "Request template file"},
+    {"section", OPT_SECTION, 's', "Config section to use (default req)"},
```
Comment (beldmit) [inline apps/req.c]: single quotes/square brackets around req?

File: doc/man1/openssl-req.pod.in [hunk 2]
```diff
@@ -328,8 +333,10 @@ argument for this option is string of hexadecimal digits.
 =head1 CONFIGURATION FILE FORMAT
 
 The configuration options are specified in the B<req> section of
-the configuration file. As with all configuration files if no
-value is specified in the specific section (i.e. B<req>) then
+the configuration file. An alternate name be specified by using the
+B<-setion> option.
```
Comment (beldmit) [inline doc/man1/openssl-req.pod.in]: Nit: typo

File: apps/req.c (hunk 1 above)
Comment (richsalz) [reply apps/req.c]: fixup commit pushed.

File: doc/man1/openssl-req.pod.in (hunk 2 above)
Comment (richsalz) [reply doc/man1/openssl-req.pod.in]: fixup commit pushed.

Comment (beldmit) [reply doc/man1/openssl-req.pod.in]: Sorry, still see '-setion'

**Comment by beldmit** [comment]
Approved if CI succeded.

**Comment by beldmit** [comment]
Squashed and merged, minor conflict fixed. Thanks!

**Pull request closed without merging**